scripts/register_sam.py

```diff
@@ -38,6 +41,13 @@ def main():
         tdr.check_api_access(source)
         tdr.check_bigquery_access(source)
 
+    public_snapshots = public_tdr.snapshot_names_by_id()
+    all_snapshots = tdr.snapshot_names_by_id()
+    require(bool(public_snapshots),
+            'The public_tdr service account cannot access any snapshots')
```

Comment:
I don't think this necessarily constitutes and error. One of the SA should be able to access snapshots. Both don't have to.

Reply:
"public_tdr service account" makes no sense given the nomenclature used in documentation and configuration. Use the established conceptual name or the actual name to avoid the confusion altogether. Please also look for places where we need to log the SA name since it is now ambiguous as to which account we're talking about. SAMClient.register_with_sam comes to mind for example.